### PR TITLE
Add documentation for S3GlueConnector.

### PR DESCRIPTION
**Pull request by vamsi-amazon**
### Description
* Added and modified documentation for s3 Glue Connector and  spark execution engine settings.
 * Documentation Links: 
       * S3 Connector: https://github.com/vamsi-amazon/sql/blob/main/docs/user/ppl/admin/connectors/s3glue_connector.rst
       * Async APIs: https://github.com/vamsi-amazon/sql/blob/main/docs/user/interfaces/asyncqueryinterface.rst

 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/user/interfaces/asyncqueryinterface.rst [hunk 1]
```diff
@@ -14,23 +14,29 @@ Async Query Interface Endpoints
...
-    plugins.query.executionengine.spark.config: '{"applicationId":"xxxxx", "executionRoleARN":"arn:aws:iam::***********:role/emr-job-execution-role","region":"eu-west-1"}'
-
-
+    plugins.query.executionengine.spark.config:
+    '{  "applicationId":"xxxxx",
+        "executionRoleARN":"arn:aws:iam::***********:role/emr-job-execution-role",
+        "region":"eu-west-1",
+        "sparkSubmitParameter": "--conf spark.dynamicAllocation.enabled=false"
+    }'
 If this setting is not configured during bootstrap, Async Query APIs will be disabled and it requires a cluster restart to enable them back again.
```
Comment (dai-chen) [inline docs/user/interfaces/asyncqueryinterface.rst:33]: just to confirm it's intentional that we check this during bootstrap and require restart if changed?

Comment (vamsi-amazon) [reply docs/user/interfaces/asyncqueryinterface.rst:33]: `region` is required during initialization. We need to figure out a way. Rest of the parameters work even after the cluster restart.

Comment (vamsi-amazon) [reply docs/user/interfaces/asyncqueryinterface.rst:33]: https://github.com/opensearch-project/sql/issues/2230
Let me work on this now.

**Review by penghuo — APPROVED** [review]

**Review by ps48 — APPROVED** [review]

File: docs/user/interfaces/asyncqueryinterface.rst (hunk 1 above)
Comment (dai-chen) [reply docs/user/interfaces/asyncqueryinterface.rst:33]: I see. I met this problem in local test. Thanks

**Pull request merged**